packages/cli/src/commands/org/resources/model/create.ts

```diff
@@ -0,0 +1,45 @@
+import {Command} from '@oclif/command';
+import {
+  IsAuthenticated,
+  Preconditions,
+} from '../../../../lib/decorators/preconditions';
+import {Trackable} from '../../../../lib/decorators/preconditions/trackable';
+import open from 'open';
+import {AuthenticatedClient} from '../../../../lib/platform/authenticatedClient';
+import {createSnapshotUrl} from '../../../../lib/platform/url';
+import {cli} from 'cli-ux';
+import dedent from 'ts-dedent';
+
+export default class Create extends Command {
+  public static description = 'Create a Snapshot Pull Model';
+
+  public static flags = {};
+
+  public static hidden = true;
+
+  @Trackable({eventName: 'org resources pull - new model'})
+  @Preconditions(IsAuthenticated())
+  public async run() {
+    await this.openPlatform();
+    cli.log(dedent`Make sure to save the snapshot for the CLI, so you can run
```

Comment:
I'm not sure I understand here 🤔 

I don't think I've seen the term 'Pull Model' before. Is that what is called a 'template' in the Swagger? https://platform.cloud.coveo.com/docs?urls.primaryName=Migration#/Snapshot/rest_organizations_paramId_snapshots_template_post

Also, I'm not sure what you mean by 'save the snapshot for the CLI'. You mean locally on your machine? Then the idea would be to pull from that local snapshot to create a snapshot in the target org?

Reply:
Hey @jpmarceau, the 'Snapshot Pull Model' (SPM) concept is specific to the CLI and without incidence to the API (kinda).

An SPM is, in essence, a JSON containing all the information to replicate a call to [
​/rest​/organizations​/{organizationId}​/snapshots​/self
](https://platform.cloud.coveo.com/docs?urls.primaryName=Migration#/Snapshot/rest_organizations_paramId_snapshots_self_post).

The _raison d'être_ of SPM is to provide a format to replicate this call at several points in time.
For example, let's say I'm doing a deployment on January 1st. I'm creating an SPM on the source organization and I use it to pull locally. The SPM contains exactly the resource I want to migrate.
Now, we're on January 3rd, I want to do exactly the same deployment again. I just have to use the SPM I saved before to pull, saving me the hassle of selecting the resource again. This is what SPMs are for, and they'll be especially useful in a CI/CD, where 'selecting the resources' would have been a real pain, if not impossible.

Reply:
Oh I think that makes sense, thanks Louis.
Does that mean that we should rephrase this string to:
`Make sure to save the resulting snapshot pull model, so you can later run "org:resources:pull -m <path/to/snapshot/pull/model.json>" to create a snapshot of the target resources in your organization.` ?

Reply:
@y-lakhdar 👀 ;)

Reply:
Does my suggestion above make sense @y-lakhdar ?

Reply:
Yes, it does. thanks!